Tests/SWBUtilTests/LinuxDistributionTests.swift

```diff
@@ -0,0 +1,776 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import Testing
+import SWBTestSupport
+import SWBUtil
+
+@Suite
+fileprivate struct LinuxDistributionTests {
+
+    /// Test helper to create a mock filesystem with specific files
+    private func withMockLinuxDistribution<T>(
+        osReleaseContent: String? = nil,
+        distributionFiles: [String: String] = [:],
+        operation: (PseudoFS) async throws -> T
+    ) async throws -> T {
+        let fs = PseudoFS()
+
+        // Create /etc directory
+        try fs.createDirectory(Path("/etc"), recursive: true)
+
+        // Add /etc/os-release if provided
+        if let content = osReleaseContent {
+            try fs.write(Path("/etc/os-release"), contents: ByteString(encodingAsUTF8: content))
+        }
+
+        // Add distribution-specific files
+        for (filePath, content) in distributionFiles {
+            try fs.write(Path(filePath), contents: ByteString(encodingAsUTF8: content))
+        }
+
+        return try await operation(fs)
+    }
+
+    /// Test parsing various /etc/os-release formats for different distributions
+    @Test
+    func detectUbuntuFromOSRelease() async throws {
```

Comment:
These tests are all duplicating the parsing code instead of testing the real implementation. If needed, I think we can just allow injecting a FileSystem into detectHostLinuxDistribution